### PR TITLE
feat(tests): making a separate crate of common tests modules

### PR DESCRIPTION
**Pull request by dsavitskiy**
Io-engine tests and benchmarks use the same common module.
Previously, benchmark used it via a symlink, which cause
problems with dependencies.
This commit also added helpers for v1 API.

Signed-off-by: Dmitry Savitskiy <dmitry.savitskiy@datacore.com>

### CODE REVIEW

**Comment by dsavitskiy** [comment]
bors try

File: io-engine-tests/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+[package]
+name = "io-engine-tests"
+description = "I/O test support library"
+version = "1.0.0"
+edition = "2018"
+authors = [
```
Comment (tiagolobocastro) [inline io-engine-tests/Cargo.toml]: nit: it's no longer recommended to add authors to crates, and it might become deprecated and removed in the future.

**Review by tiagolobocastro — APPROVED** [review]

**Review by mtzaurus — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -7,6 +7,7 @@ members = [
 	"libnvme-rs",
 	"io-engine",
     "io-engine-bench",
+    "io-engine-tests",
```
Comment (Abhinandan-Purkait) [inline Cargo.toml]: nit: is the indentation incorrect here?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: io-engine-tests/Cargo.toml (hunk 1 above)
Comment (dsavitskiy) [reply io-engine-tests/Cargo.toml]: I'll remove it.

File: Cargo.toml (hunk 2 above)
Comment (dsavitskiy) [reply Cargo.toml]: It's okay on my machine. Probably, a tab is here. I'll re-format the file.

**Comment by dsavitskiy** [comment]
bors merge

**Pull request merged**